env/types.go

```diff
@@ -0,0 +1,185 @@
+	if !envVar.IsSet {
+		envVar.Value = defaultValue
+		return envVar
+	}
+	value, err := strconv.ParseBool(envVar.Raw)
+	if err != nil {
+		panic(fmt.Sprintf("unable to parse %s - should be 'true' or 'false'", envVar.Key))
+	}
+	envVar.Value = value
+	return envVar
+}
+
+func newIntEnvVar(defaultValue int, keys ...string) IntEnvVar {
+	envVar := IntEnvVar{eVar: newEVar(keys...)}
+	if !envVar.IsSet {
+		envVar.Value = defaultValue
+		return envVar
+	}
+	value, err := strconv.ParseInt(envVar.Raw, 0, 0)
+	if err != nil {
+		panic(fmt.Sprintf("unable to parse %s - does not seem to be an int", envVar.Key))
+	}
+	envVar.Value = int(value)
+	return envVar
+}
+
+func newStringEnvVar(defaultValue string, keys ...string) StringEnvVar {
+	envVar := StringEnvVar{eVar: newEVar(keys...)}
+	if !envVar.IsSet {
+		envVar.Value = defaultValue
+		return envVar
+	}
+	envVar.Value = envVar.Raw
+	return envVar
+}
+
+func newSliceEnvVar(defaultValue []string, keys ...string) SliceEnvVar {
+	envVar := SliceEnvVar{eVar: newEVar(keys...)}
+	if !envVar.IsSet {
+		envVar.Value = defaultValue
+		return envVar
+	}
+	val := strings.Split(envVar.Raw, ",")
+	for i := range val {
+		val[i] = strings.TrimSpace(val[i])
+	}
+	envVar.Value = val
+	return envVar
+}
+
+func newMapEnvVar(defaultValue map[string]interface{}, keys ...string) MapEnvVar {
+	envVar := MapEnvVar{eVar: newEVar(keys...)}
+	if !envVar.IsSet {
+		envVar.Value = defaultValue
+		return envVar
+	}
+	valItems := strings.Split(envVar.Raw, ",")
+	for i := range valItems {
+		valItems[i] = strings.TrimSpace(valItems[i])
+	}
+	val := map[string]interface{}{}
+	for _, item := range valItems {
+		itemArr := strings.Split(item, "=")
+		if len(itemArr) == 2 {
+			itemValue := itemArr[1]
+			if len(itemValue) > 0 && itemValue[0] == '$' {
+				itemValue = os.Getenv(itemValue[1:])
+			}
+			val[itemArr[0]] = itemValue
+		}
+	}
+	envVar.Value = val
+	return envVar
+}
+
+// For use in if's
+
+func (e *BooleanEnvVar) Tuple() (bool, bool) {
+	return e.Value, e.IsSet
+}
+func (e *IntEnvVar) Tuple() (int, bool) {
+	return e.Value, e.IsSet
+}
+func (e *StringEnvVar) Tuple() (string, bool) {
+	return e.Value, e.IsSet
+}
+func (e *SliceEnvVar) Tuple() ([]string, bool) {
+	return e.Value, e.IsSet
+}
+func (e *MapEnvVar) Tuple() (map[string]interface{}, bool) {
+	return e.Value, e.IsSet
+}
+
+// For fallback support
+
+func (e *BooleanEnvVar) ValueIfSetOr(fallback bool) bool {
+	if e.IsSet {
+		return e.Value
+	}
+	return fallback
+}
+func (e *IntEnvVar) ValueIfSetOr(fallback int) int {
+	if e.IsSet {
+		return e.Value
+	}
+	return fallback
+}
+func (e *StringEnvVar) ValueIfSetOr(fallback string) string {
+	if e.IsSet {
+		return e.Value
+	}
+	return fallback
+}
+func (e *SliceEnvVar) ValueIfSetOr(fallback []string) []string {
+	if e.IsSet {
+		return e.Value
+	}
+	return fallback
+}
+func (e *MapEnvVar) ValueIfSetOr(fallback map[string]interface{}) map[string]interface{} {
+	if e.IsSet {
+		return e.Value
+	}
+	return fallback
+}
```

Comment:
This seems a bit overkill for me. We only use `func (e *BooleanEnvVar) ValueIfSetOr(fallback bool) bool` once, so not sure we want to add all this complexity just for that.

Reply:
Yeah you're right, I wrote those methods first as utils, but if they are used only once, is better to remove them and change the callers logic.

Reply:
Done